lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -254,6 +255,18 @@ function collateResults(localConsole, actual, expected) {
     return makeComparison(auditName + ' audit', actualResult, expectedResult);
   });
 
+  /** @type {Comparison[]} */
+  const requestCountAssertion = [];
+  if (expected.networkRequests) {
+    const url = new URL(actual.lhr.requestedUrl);
+    const urlPath = url.pathname + url.search;
+    requestCountAssertion.push(makeComparison(
+      'Request count',
+      server.getRequestUrls(urlPath),
+      expected.networkRequests
```

Comment:
i think it'd be good to exercise the actual assertions within the PR.

Reply:
I added some expectations to perf.

Reply:
> I added some expectations to perf.

dbw might be one of the best to add expectations to since the page tries to be bad at everything and the smoke test runs the full default config. It's also just the one smoke test, so shouldn't be much work to add.

Reply:
dbw is two smoke tests run in parallel, one smoke test is an external URL.

The smoke test should still work if we adjust our requirement from "Tests run synchronously" to "Only one test using the static server at a time". Definitely doable, but it might add a bit of complexity.

Reply:
Ah, yeah, I meant just the first one anyways. Agreed it's not worth the complexity (at least for now).

Reply:
That is unfortunately still a problem. Under the current implementation, the expectation would be pruned/throw because dbw is run in parallel. We need to change the definition (adding complexity) for the expectation to be actually be asserted.

Reply:
Oh, I see, you're talking about `runSerially` so we've been talking about slightly different things this whole time, sorry :/ 

You have the `isSync` value based on `concurrency`, which is the combination of `runSerially` with the `--jobs` flag/option:

https://github.com/GoogleChrome/lighthouse/blob/802947d2d4b97b03912b9756c8b4607568523ee0/lighthouse-cli/test/smokehouse/smokehouse.js#L55-L56

so it'll be pruned in a the usual local runs, but in CI we run with `-j=1` for vm performance reasons (and maybe still coverage issues?) so `concurrency` will still be 1 and it shouldn't be pruned there.

Reply:
Now that I write that, maybe it'll be annoying when editing dbw to only find out about failures in CI unless you happen to run with `-j=1`.

If ever there was a smoke test perfect for #11506 it would be dbw, though. Should we just `runSerially: true` and take the hit on slower smoke tests but keep things simpler?

Reply:
> Should we just `runSerially: true` it and take the hit on slower smoke tests but keep things simpler?

That's what I'm thinking, it shouldn't be too bad because there are only two tests.

Reply:
Just added dbw.